explorer-api/src/country_statistics/mod.rs

```diff
@@ -51,17 +52,35 @@ impl CountryStatistics {
 
         info!("Locating mixnodes...");
         for (i, bond) in mixnode_bonds.iter().enumerate() {
-            match locate(&bond.mix_node.host).await {
+            match locate(&bond.1.bond.mix_node.host).await {
                 Ok(location) => {
                     let country_code = map_2_letter_to_3_letter_country_code(&location);
                     *(distribution.entry(country_code)).or_insert(0) += 1;
 
+                    let three_letter_iso_country_code =
+                        map_2_letter_to_3_letter_country_code(&location);
+
                     trace!(
                         "Ip {} is located in {:#?}",
-                        bond.mix_node.host,
-                        map_2_letter_to_3_letter_country_code(&location)
+                        bond.1.bond.mix_node.host,
+                        three_letter_iso_country_code,
                     );
 
+                    self.state
+                        .inner
+                        .mix_nodes
+                        .set_location(
+                            &bond.1.bond.mix_node.identity_key,
+                            Location {
```

Comment:
To collapse it a little bit, maybe we could have a constructor on `Location` that takes `three_letter_iso_country_code` and `GeoLocation` as arguments? Then we'd just have `Location::new(three_letter_iso_country_code, location)`

Reply:
Refactored it a bit to simplify with the constructor